py/modsys.c

```diff
@@ -60,15 +60,24 @@ STATIC const MP_DEFINE_STR_OBJ(mp_sys_version_obj, "3.4.0; " MICROPY_BANNER_NAME
 
 // version_info - Python language version that this implementation conforms to, as a tuple of ints
 #define I(n) MP_OBJ_NEW_SMALL_INT(n)
-// TODO: CPython is now at 5-element array, but save 2 els so far...
+// TODO: CPython is now at 5-element array (major, minor, micro, releaselevel, serial), but save 2 els so far...
 STATIC const mp_obj_tuple_t mp_sys_version_info_obj = {{&mp_type_tuple}, 3, {I(3), I(4), I(0)}};
 
 // sys.implementation object
 // this holds the MicroPython version
 STATIC const mp_obj_tuple_t mp_sys_implementation_version_info_obj = {
     {&mp_type_tuple},
-    3,
-    { I(MICROPY_VERSION_MAJOR), I(MICROPY_VERSION_MINOR), I(MICROPY_VERSION_MICRO) }
+    4,
+    {
+        I(MICROPY_VERSION_MAJOR),
+        I(MICROPY_VERSION_MINOR),
+        I(MICROPY_VERSION_MICRO),
+        #if MICROPY_VERSION_PRERELEASE
+        MP_ROM_QSTR(MP_QSTR_preview),
+        #else
+        MP_ROM_QSTR(MP_QSTR_),
```

Comment:
CPython seems to use "final" here... maybe we can do the same??  Or maybe that's just a big waste of bytes.

Reply:
I actually had it as "final" initially for exactly that reason, and "beta" instead of "preview" (CPython supports "alpha", "beta", "candidate", "final").

But I thought matching "preview" (to the use everywhere else) was sensible and therefore no reason to match "final".

Reply:
OK, sounds good.